### PR TITLE
Pf upstream changes

### PR DESCRIPTION
**Pull request by petefoth**
These are changes I made to make my unofficial and custom builds of LineageOS for Microg for devices not officially supported by LOS.

The Custom builds are intended to provide 'out-of-the-box' similar functionality to /e/OS (pre-Avanced Privacy), with /e/'s forked apps replaced by their upstreams.

More details in my project wiki https://codeberg.org/petefoths-projects/unofficial-l4m-builds/wiki

(Apologies for the first three commits: I haven't worked out how to pull in your changes to my form **without** creating a merge commit)

### CODE REVIEW

File: extra/Android.mk [hunk 1]
```diff
@@ -1,12 +1,33 @@
...
 # additional F-Droid repos
 include $(CLEAR_VARS)
-LOCAL_MODULE := additional_repos.xml
+LOCAL_MODULE := extendrom_additional_repos.xml
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE_CLASS := ETC
 LOCAL_MODULE_PATH := $(TARGET_OUT_VENDOR_ETC)/org.fdroid.fdroid
-LOCAL_SRC_FILES := $(LOCAL_MODULE)
+LOCAL_OVERRIDES_PACKAGES := additional_repos.xml
+LOCAL_SRC_FILES := additional_repos.xml
```
Comment (steadfasterX) [inline extra/Android.mk:30]: why is that needed? can't we just keep `additional_repos.xml`? I also wondering about the LOCAL_OVERRIDES_PACKAGES in that context.. does that mean you want to keep it "backwards"  compatible for those having `additional_repos.xml` specified in their config? If the above works like that it would be ok I just want to understand how that works or what the idea is behind

File: extra/Android.mk [hunk 2]
```diff
@@ -526,3 +547,78 @@ LOCAL_MODULE_CLASS := ETC
...
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_CLASS := APPS
+LOCAL_DEX_PREOPT := false
+LOCAL_MODULE_SUFFIX := .apk
+LOCAL_OVERRIDES_PACKAGES := Snap
+include $(BUILD_PREBUILT)
+
+# override lineageOS Camera2
+include $(CLEAR_VARS)
+LOCAL_MODULE := noLOSSnap
```
Comment (steadfasterX) [inline extra/Android.mk:589]: is that a typo or should it really be `LOCAL_MODULE := noLOSSnap` here like in the above block? If that is not a typo: won't that cause conflicts? I would assume more something like `noLOScamera2` here tbh..

File: extra/bromite_repos.xml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
...a864886f70d01010105000382010f003082010a02820101009a8d2a5336b0eaaad89ce447828c7753b157459b79e3215dc962ca48f58c2cd7650df67d2dd7bda0880c682791f32b35c504e43e77b43c3e4e541f86e35a8293a54fb46e6b16af54d3a4eda458f1a7c8bc1b7479861ca7043337180e40079d9cdccb7e051ada9b6c88c9ec635541e2ebf0842521c3024c826f6fd6db6fd117c74e859d5af4db04448965ab5469b71ce719939a06ef30580f50febf96c474a7d265bb63f86a822ff7b643de6b76e966a18553c2858416cf3309dd24278374bdd82b4404ef6f7f122cec93859351fc6e5ea947e3ceb9d67374fe970e593e5cd05c905e1d24f5a5484f4aadef766e498adf64f7cf04bddd602ae8137b6eea40722d0203010001a321301f301d0603551d0e04160414110b7aa9ebc840b20399f69a431f4dba6ac42a64300d06092a864886f70d01010b0500038201010007c32ad893349cf86952fb5a49cfdc9b13f5e3c800aece77b2e7e0e9c83e34052f140f357ec7e6f4b432dc1ed542218a14835acd2df2deea7efd3fd5e8f1c34e1fb39ec6a427c6e6f4178b609b369040ac1f8844b789f3694dc640de06e44b247afed11637173f36f5886170fafd74954049858c6096308fc93c1bc4dd5685fa7a1f982a422f2a3b36baa8c9500474cf2af91c39cbec1bc898d10194d368aa5e91f1137ec115087c31962d8f76cd120d28c249cf76f4c70f5baa08c70a7234ce4123be080cee789477401965cfe537b924ef36747e8caca62dfefdd1a6288dcb1c4fd2aaa6131a7ad254e9742022cfd597d2ca5c660ce9e41ff537e5a4041e37</item>
```
Comment (steadfasterX) [inline extra/bromite_repos.xml:32]: duplicate with `extra/microg_repos.xml` so they should be removed from here

File: extra/microg_repos.xml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    </string-array>
+</resources>
```
Comment (steadfasterX) [inline extra/microg_repos.xml:35]: I like the idea of splitting the extra repos but for microG you should then remove:

1. https://github.com/sfX-android/android_vendor_extendrom/blob/main/extra/additional_repos.xml#L5-L17
2. https://github.com/sfX-android/android_vendor_extendrom/blob/main/extra/additional_repos.xml#L34-L47

**Review by steadfasterX — CHANGES_REQUESTED** [review]
thanks for your contribution! Please check my comments and let me know if you have any questions - either here or on my TG group ;)

File: extra/Android.mk (hunk 1 above)
Comment (petefoth) [reply extra/Android.mk:30]: Yes. I wanted it to work for people who want both the MicroG repos **and** the Bromite repos. 

I also think it needed renaming because my builds are picking up MicroG (and F-Droid etc.) from the [`lineageos4microg/android_vendor_partner_gms` repo](https://github.com/lineageos4microg/android_vendor_partner_gms), which includes a `additional_repos.xml` module, and the build complained about the duplication

File: extra/Android.mk (hunk 2 above)
Comment (petefoth) [reply extra/Android.mk:589]: Yes - it should be`noLOScamera2`

File: extra/bromite_repos.xml (hunk 3 above)
Comment (petefoth) [reply extra/bromite_repos.xml:32]: OK
It will take me some time to deal with this, as I have never tried to modify a branch once its been submitted as an MR. I'll try wand work out what I need to do next time I get a couple of hours spare to work on this

File: extra/microg_repos.xml (hunk 4 above)
Comment (petefoth) [reply extra/microg_repos.xml:35]: OK. See my previous comment

Comment (petefoth) [reply extra/microg_repos.xml:35]: Before I start changing the bits that deal with `additional_repos.xml` I've been looking into the [F_Droid Android client code](https://gitlab.com/fdroid/fdroidclient). What I found is that the function [`DBHelper.loadAdditionalRepos()`](https://gitlab.com/fdroid/fdroidclient/-/blob/master/app/src/main/java/org/fdroid/fdroid/data/DBHelper.java#L135) looks in the `etc` subdirectory ofa number of different directories (`/system, /product, /vendor, /odm, /oem`) for a file named `additional_repos.txt`, and adds the repos defined in that file.

So, if we want to be able to add different sets of repos for different target modules (microg, bromite, our original additional repos), then we have to combine all those lists of repos **into a single file** named `additional_repos.xml`. This would involve even more code in the processing of `packages.txt` which is well beyond my ability to implement in a shell function, and would (might) also involve other changes in this project. 

For my builds, I want to add the Bromite repos **and** the microG repos (so that the built-in microG from `vendor_partner_gms`can be updated via F-Droid). So I've bodged it by putting **both** in `bromite_repos.xml`, and changing `extra/Android.mk` so that whichever you choose gets written out to `/vendor/etc/org.fdroid.fdroid`

This works for my builds, but it is going to break if someone tries to include more than one of `additional_repos.xml`, `MicrogGmsCore` and `Bromite` in `EXTENDROM_PACKAGES`:  the version of `additional_repos.xml`  that gets written to `/vendor/etc/org.fdroid.fdroid` will be the version "belonging" to the last of those modules to be processed in `get_prebuilts.sh`. (Or maybe we'll get a build error - I'm a bit out of my depth here!)

For now, I think the best way forward is
1. I will submit a new PR, with only the changes that don't involve messing with `additional_repos.xml`
2. I will continue to bodge my builds, with all the repos I need in `bromite.xml`
3. When / if we come up with a way of adding different lists of additional repos to the same `additional_repos.xml` I can test it by "un-bodging" my changes

If you're happy with that, I'll make a start on the new PR

File: extra/bromite_repos.xml (hunk 3 above)
Comment (steadfasterX) [reply extra/bromite_repos.xml:32]: you simply fix the code and push to your same branch and it gets updated here :)

File: extra/Android.mk (hunk 2 above)
Comment (petefoth) [reply extra/Android.mk]: Commited the change

File: extra/bromite_repos.xml (hunk 3 above)
Comment (petefoth) [reply extra/bromite_repos.xml]: Done

File: extra/microg_repos.xml (hunk 4 above)
Comment (petefoth) [reply extra/microg_repos.xml:35]: I have changed things so that there is no duplication between `extendrom_additional_repos`, `bromite_repos` and `microg_repos`. But these are still mutually exclusive - it is not possible to have combinations (like `bromite_repos` **and** `microg_repos`.
 I will address that with #26 in a future PR, so I think this can be resolved now

**Comment by petefoth** [comment]
Closing this PR

Changes will be made in [PR ` pf l4m changes #27`](https://github.com/sfX-android/android_vendor_extendrom/pull/27)
